packages/grafana-data/src/types/annotations.ts

```diff
@@ -86,24 +90,35 @@ export type AnnotationEventMappings = Partial<Record<keyof AnnotationEvent, Anno
 export interface AnnotationSupport<TQuery extends DataQuery = DataQuery, TAnno = AnnotationQuery<TQuery>> {
   /**
    * This hook lets you manipulate any existing stored values before running them though the processor.
-   * This is particularly helpful when dealing with migrating old formats.  ie query as a string vs object
+   * This is particularly helpful when dealing with migrating old formats.  ie query as a string vs object.
    */
-  prepareAnnotation?(json: any): TAnno;
+  prepareAnnotation?(json: AnnotationQuery<TQuery>): AnnotationQuery<TQuery>;
```

Comment:
The argument json can be old annotation data but that should still conform to AnnotationQuery as it can have any key: value.

Reply:
The 'any' was required because in the wild, old annotation objects were of almost any form.

We may be able to do: `TAnno | string | object`, but I am not sure that is any better than just `any` 🤷

Reply:
When looking at the code it seemed like other parts of the code handling legacy annotations also type it as `AnnotationQuery<TQuery>`. I guess the comment then wasn't very clear as `ie query as a string vs object` meant to be `ie json as a string vs object`? Thought then the name of the arg would be a bit misleading too.